protected/humhub/components/ModuleManager.php

```diff
@@ -213,23 +216,106 @@ public function register($basePath, $config = null)
         Yii::$app->setModule($config['id'], $moduleConfig);
 
         // Register Event Handlers
-        if (isset($config['events'])) {
-            foreach ($config['events'] as $event) {
-                $eventClass = $event['class'] ?? $event[0];
-                $eventName = $event['event'] ?? $event[1];
-                $eventHandler = $event['callback'] ?? $event[2];
-                $eventData = $event['data'] ?? $event[3] ?? null;
-                $eventAppend = filter_var($event['append'] ?? $event[4] ?? true, FILTER_VALIDATE_BOOLEAN);
-                if (method_exists($eventHandler[0], $eventHandler[1])) {
-                    Event::on($eventClass, $eventName, $eventHandler, $eventData, $eventAppend);
-                }
-            }
-        }
+        $this->registerEventHandlers($basePath, $config);
 
         // Register Console ControllerMap
         if (Yii::$app instanceof ConsoleApplication && !(empty($config['consoleControllerMap']))) {
             Yii::$app->controllerMap = ArrayHelper::merge(Yii::$app->controllerMap, $config['consoleControllerMap']);
         }
+
+        return $config['id'];
+    }
+
+    /**
+     * @throws InvalidConfigException
+     */
+    protected function registerEventHandlers(string $basePath, array &$config): void
+    {
+        $events = $config['events'] ?? null;
+        $strict = $config['strict'] ?? false;
+
+        if (empty($events)) {
+            return;
+        }
+
+        $error = static function (string $message, bool $throw = false) use (&$config, $basePath) {
+            $message = "Configuration at $basePath has an invalid event configuration: $message";
+
+            if ($throw) {
+                throw new InvalidConfigException($message);
+            }
+
+            Yii::warning($message, $config['id']);
+        };
+
+        if (!ArrayHelper::isTraversable($events)) {
+            $error('events must be traversable', $strict);
+            return;
+        }
+
+        $getProperty = static function ($event, &$var, string $property, int $index, bool $throw = false) use ($error): bool {
+
+            $var = $event[$property] ?? $event[$index] ?? null;
+
+            if (empty($var)) {
+                $error("required property '$property' missing!", $throw);
```

Comment:
@luke-, I guess I should also update this line here, right?

Reply:
Yes, would be great.